test.js

```diff
@@ -148,14 +148,47 @@ function testStep(context, conversationId, userMessage, expectedReplies, timeout
         });
 }
 
+// Replacement method for chai's deep.equal
+function deepEqual(expected, actual) {
+    // Test using regex for attributes starting with regex keyword 
+    if ((typeof expected === "string" && expected.startsWith("regex:"))) {
+        const regex = new RegExp(expected.replace("regex:", "").trim());
+        if (!regex.test(actual))
+            throw "Actual value doesn't match provided regex, Regex: " + regex.toString() + ", Actual: " + JSON.stringify(actual);
+        else return true;
+    }
+    // Regular test for other values
+    else if (expected === actual)
+        return true;
+    else if ((typeof expected === "object" && expected !== null) && (typeof actual === "object" && actual !== null)) {
+        for (var prop in expected) {

```

Comment:
Should also check that number of attributes number to ensure deep equality.

Reply:
This will conflict with what we did in the transformation script, kindly consult with Amir on how/when we should use this deepEqual method and get back to me.